Update the docs release playbook

zacharysarah opened the pull request:
This PR:
- Adds new information
- Removes stale information
- closes https://github.com/kubernetes/website/issues/8143
- ref #192

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+## Introduction
 
-This doc contains an outline of tasks to be completed during a Kubernetes.io docs release cycle for a Kubernetes minor release, e.g. v1.9.
+Kubernetes releases minor versions on a quarterly cadence. For example, in 2017, the releases were: v1.6 (Q1/Jan-Mar), v1.7 (Q2/Apr-Jun), v1.8 (Q3/Jul-Sep), and v1.9 (Q4/Oct-Dec).
```
Might be better as a bullet list

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+Kubernetes releases minor versions on a quarterly cadence. For example, in 2017, the releases were: v1.6 (Q1/Jan-Mar), v1.7 (Q2/Apr-Jun), v1.8 (Q3/Jul-Sep), and v1.9 (Q4/Oct-Dec).
 
-## Quarterly release cycle
+SIG Docs is part of the Kubernetes release process: its [maintainers](https://github.com/orgs/kubernetes/teams/sig-docs-maintainers) support documentation created/updated for releases and published on the [kubernetes website](https://kubernetes.io/).
```
I've only ever seen it called sig-docs, not SIG Docs.

Why not "created or updated"?

tao12345666333 replied:
Maybe it should be called sig-docs.

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
-## Managing the cycle
-
-The designated "release meister" (aka Docs Lead), voluntold from [@kubernetes/sig-docs-maintainers](https://github.com/orgs/kubernetes/teams/sig-docs-maintainers), is responsible for managing the process on the docs side and releasing the documentation on [kubernetes.io](https://kubernetes.io/) when the Kubernetes binary release goes out.
+The designated "release meister" (aka Docs Lead), voluntold from [@kubernetes/sig-docs-maintainers](https://github.com/orgs/kubernetes/teams/sig-docs-maintainers), is responsible for managing the docs release  for new and updated documentation [kubernetes.io](https://kubernetes.io/) during quarterly releases.
```
s/aka//

s/is responsible for managing/manages

nit: extra space

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
-    *   [Kubernetes Features OSS tracking board (1.7 release)](https://docs.google.com/spreadsheets/d/1IJSTd3MHorwUt8i492GQaKKuAFsZppauT4v1LJ91WHY/edit?usp=sharing)
-    *   [Kubernetes Features OSS tracking board (1.8 release)](https://docs.google.com/spreadsheets/d/1AFksRDgAt6BGA3OjRNIiO3IyKmA-GU7CXaxbihy48ns/edit#gid=0)
-    *   Not sure whether there will be a tracking spreadsheet for 1.9
+*   k8s.io (docs) GitHub repo - [kubernetes/website](https://github.com/kubernetes/website)
```
Would it be better to use kubernetes.io instead of k8s.io as the "official website" in all of these list items?

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
-### Generate reference docs
+## Start the release cycle
+
+Release cycles usually begin 1-2 weeks after the previous cycle completes.
```
Link to release calendar / schedule?

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+1. Get connected.
+
+- Join the Kubernetes [#sig-release Slack channel](https://kubernetes.slack.com/messages/C2C40FMNF/).
```
I think that the sub-bullet dashes need to be indented further to align with the 'G' in 'Get connected' to not break up the outer list. Actually I'm not sure if you intended them to be sub-bullets or you just decided to switch from numbered to bullet after the first point. Regardless something is wrong.

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+    - [kubernetes-sig-leads](https://groups.google.com/forum/#!forum/kubernetes-sig-leads)
+    - [kubernetes-dev](https://groups.google.com/forum/#!forum/kubernetes-dev)
+
+- Make sure you're included in the release team docs, for example: [release 1.11](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.11/release_team.md).
```
Perhaps by submitting a PR to the correct document?

mistyhacks commented on release-process-documentation/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+### Docs starting tasks
+
+1. Create a pull request for the release.
```
Again sub-bullet problems here.

mistyhacks commented:
Oh sorry, your mention of it in Slack is what prompted me to review. Sorry for the mix-up.

zacharysarah commented:
@mistyhacks No worries! 💛 I appreciate your review, just want to make sure your feedback gets the most action in response.

tpepper commented:
Where's this stand?  Would it be possible to get this finished/merged in the next couple weeks as 1.11 finishes and 1.12 starts so there's solid guidance for 1.12?

justaugustus commented:
Hey y'all, could we have the 1.12 / 1.13 Docs peeps pick this up and get it merged in advance of the 1.13 release cycle.
We need to integrate the review suggestions and adapt it to fit the new dir structure.

There also seem to be three docs about writing docs. Could we merge those together?
Additionally, any information that is written for the audience that would be writing enhancement documentation should be moved to a relevant guide in `k/community`.

/assign @zparnold @jimangel @tfogo

zacharysarah commented:
@zparnold @justaugustus This PR is an incomplete work in progress, abandoned months ago. Are you sure you want to merge it as is? 

I can’t pick this back up again right now or in the near future. I’m OK with merging, however, if it’s a roadblock to iterative improvements like I think @zparnold is suggesting. (Is that what you’re suggesting? :bow tie:)

justaugustus commented:
@zparnold -- Sorry for any confusion...
I was suggesting pulling in Zachary's work and iterating on top of it. You could do this:
```
cd `local-dir-for-sig-release-repo`
git remote add zacharysarah git@github.com:zacharysarah/sig-release.git
git remote update
git checkout zacharysarah/docs-playbook-update
git checkout -b <some-branch-name>
git rebase -i HEAD~3 # this will start an interactive rebase of Zach's commits. Squash the bottom 2.
git rebase <remote-name-of-k/sig-release>/master
# Fix the rebase
# Continue iterating on the handbook from here
```

zparnold commented:
Righto! Can do. Likely tomorrow though. 🛩 😄

justaugustus commented:
Hehe, enjoy what's left of the weekend!

justaugustus commented:
Bump

mistyhacks commented:
/unassign

AishSundar commented:
Hi @zparnold @tfogo making sure this is still in your radar to get merged before 1.13 begins? Even if we can get the approved portions before the cycle start, we can work on incremental updates after that. Thanks !

AishSundar commented:
Also should we close https://github.com/kubernetes/sig-release/issues/192 in favor of this issue?

tfogo commented:
Note, this file got moved into `release-team/role-handbooks/documentation-guides`

I agree that we can take this as a starting point and take time during 1.13 to ensure its accuracy and improve where possible.

zacharysarah commented:
For the sake of moving on for 1.13, I've gotten this PR to a mergeable state.

@AishSundar It's a partial start, but I wouldn't say it's enough to close the issue.

@tfogo Thanks for the note--I've rebased onto `<this-repo>/master` and squashed the commits.

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+- Make sure you're included in the release team docs, for example: [release 1.11](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.11/release_team.md).
+
+- Review (and correct if necessary) the proposed timeline for the release, for example: [release 1.11](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.11/release-1.11.md).
```
is this the right landing page for this bullet? FWIW we are not using .md anymore and exclusively using only the README

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+- Watch for the release feature tracking spreadsheet.
+
+    Early in the cycle, the release manager opens a feature tracking spreadsheet (for example, [the 1.8 release spreadsheet](https://docs.google.com/spreadsheets/d/1AFksRDgAt6BGA3OjRNIiO3IyKmA-GU7CXaxbihy48ns/edit#gid=0)). The spreadsheet contains important information for docs:
```
s/release manager/enhancements lead

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+    A tracking comment helps track the identity and status of component PRs in the megabranch.
+
+    For example: https://github.com/kubernetes/website/pull/8046#issuecomment-380226165
```
can we make this a link please

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+    Keep the feature tracking spreadsheet up to date with review progress and merge status for each documentation PR. For example: [Kubernetes Features OSS tracking board (1.11 release)](https://docs.google.com/spreadsheets/d/16N9KSlxWwxUA2gV6jvuW9N8tPRHzNhu1-RYY4Y0RZLs/edit#gid=0)
+
+- Update documentation PR progress in the megabranch PR's tracking comment. For example, [Release 1.11 (tracking comment)](https://github.com/kubernetes/website/pull/8046#issuecomment-380226165)
```
Sorry if this isnt clear just to me. Looking ay the specific comment linked here I dont see any status update except for the placeholder text. Is there any other PR or comment we can point to here that explains this point better?

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+- Periodically merge `master` into the current release branch.
+
+For example, during the v1.11 release cycle, you must periodically merge `master` into `release-1.10`.
```
should this be release-1.11?

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+
+For example, during the v1.11 release cycle, you must periodically merge `master` into `release-1.10`.
+
+Periodic merges must continue throughout the release cycle, until the pending release (for example, `release-1.11`) merges to `master`. This is to ensure that the current release branch remains up to date, and to preserve the current branch as a snapshot when the new version is released. Periodic maintenance also helps avoid excessive merge conflicts between `master` and the current branch at the ...
```
you mean we do this until the code thaw period ? i.e., as long as PRs from master is accepted into the release?

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
+*   Triage:
     *   Decide what must go out with release, what can be published just after the release.
     *   Determine how much effort will be required to review the doc(s).
         *   Is it a complex feature needing specific reviewers?
```
Can you plz add a point the average turnaround time for translations that we need to factor in for last minute PRs

AishSundar commented on release-team/role-handbooks/documentation-guides/update-release-docs-new.md:
```diff
@@ -1,88 +1,180 @@
 
-*   Triage: 
+*   Triage:
     *   Decide what must go out with release, what can be published just after the release.
```
Do we have a exception process in place for features with late incoming doc PRs? If not we can work on something for 1.13. 

Basically I would like to somehow emphasize in the verbiage here that we will and should punt features that are delayed on critical doc PRs that need to go with the release.  We should treat the final doc PR deadline as a hard one one

tfogo commented:
Thanks @zacharysarah! Let's move this off your plate for now and we can create a follow-up PR to make improvements. 

@AishSundar we can work on your feedback in the follow-up PR.

/lgtm
